components/src/core/ToolbarMenu/ToolbarMenu.tsx

```diff
@@ -0,0 +1,259 @@
+            for (const group of menuGroups) {
+                for (const item of group.items) {
+                    const onClick = item.onClick;
+                    if (onClick) {
+                        item.onClick = (e: React.MouseEvent<HTMLElement, MouseEvent>): void => {
+                            onClick(e);
+                            closeMenu();
+                        };
+                    }
+                }
+            }
+        }
+    }, [menuGroups]);
+
+    const getMenu = useCallback(() => {
+        if (menu && Boolean(anchorEl)) {
+            return React.cloneElement(menu, {
+                anchorEl: anchorEl,
+                open: Boolean(anchorEl),
+                onClose: closeMenu,
+                ...menu.props,
+            });
+        }
+        if (menuGroups && Boolean(anchorEl)) {
+            return (
+                <Menu
+                    getContentAnchorEl={null}
+                    anchorOrigin={{ vertical: 'bottom', horizontal: rtl ? 'right' : 'left' }}
+                    transformOrigin={{ vertical: 'top', horizontal: rtl ? 'right' : 'left' }}
+                    anchorEl={anchorEl}
+                    open={Boolean(anchorEl)}
+                    onClose={(): void => closeMenu()}
+                    {...MenuProps}
+                    MenuListProps={{ style: { padding: 0 } }}
+                >
+                    {!menu &&
+                        menuGroups.map((group: ToolbarMenuCompGroup, index: number) => (
+                            <div className={defaultClasses.navGroups} key={index}>
+                                <DrawerNavGroup
+                                    divider={false}
+                                    hidePadding={true}
+                                    itemIconColor={group.iconColor}
+                                    itemFontColor={group.fontColor}
+                                    title={group.title}
+                                    items={group.items.map(
+                                        (item: ToolbarMenuCompItem, itemIndex: number): NavItem =>
+                                            Object.assign({ itemID: itemIndex.toString() }, item)
+                                    )}
+                                />
+                            </div>
+                        ))}
+                </Menu>
+            );
+        }
+    }, [menuGroups, menu, anchorEl, MenuProps, defaultClasses]);
+
+    return (
+        <>
+            <Typography
+                ref={composeRefs(ref, anchor)}
+                aria-haspopup="true"
+                className={clsx(
+                    defaultClasses.root,
+                    classes.root,
+                    menuGroups || menu ? defaultClasses.cursorPointer : ''
+                )}
+                component={'span'}
+                data-test={'wrapper'}
+                onClick={(): void => {
+                    openMenu(anchor.current);
+                }}
```

Comment:
When you spread the Typography props in here, you'll need to be careful and order the props properly so none of the behavior gets broken:

```
<Typography
    // props that we set, but we don't mind / won't break if the user overrides (e.g., aria-haspopup)
    {...otherTypographyProps}
    // props that we will merge with what the user provides (e.g., className)
    // props that we set that we will not allow the user to override because it will break things (e.g., onClick)
/>
```